### DISCUSSION TITLE
Testing Subqueries/Mutations of a top level query (codegen for subqueries/mutations)

### DISCUSSION THREAD

**Discussion by seanscal** [Netflix/dgs-framework#528] [Q&A] [answered]
Hey all, was wondering if it is possible to test this query currently - I am not seeing any way to do so using current codegen without using plain strings (which lead to a lot of time wasted debugging small syntax errors. My issue is i cannot seem to propagate downwards any of the variables i enter in, so we cannot generate a real `signUp` query at all here let alone figure out how to test from there. 

```graphql
mutation userSignup {
  topLevel {
    signUp(
      input: {first: "x", last: "y"}}
    ) {
      success
      id
    }
  }
}
```
```graphql
type Mutation {
    topLevel: MyTopLevelMutation!
}

type MyTopLevelMutation {
    # Signups a new user
    signUp(input: UserInput!): UserPayload!
    
    #...more mutations
}

type UserInput {
   first: String!
   last: String!
}

type UserPayload {
    success: Boolean!
    id: ID!
}

```

**Comment by paulbakker** [comment, marked as answer]
This was recently fixed in codegen. Please try again with the latest release(https://github.com/Netflix/dgs-codegen/releases)!

**Reply by seanscal** [reply]
amazing! thanks :)

**Comment by sriramgopalan-awx** [comment]
can you post on how you implemented the data fetchers for these?
